src/weather_gen.cpp

```diff
@@ -27,23 +27,37 @@ int weather_generator::current_winddir = 1000;
 w_point weather_generator::get_weather( const tripoint &location, const time_point &t,
                                         unsigned seed ) const
 {
-    static constexpr int weather_cache_limit = 10000000;
+    // Granularity of time: weather changes only once in 30 minutes
+    static constexpr int t_mult = 300;
+    // Granularity of coordinates: weather is the same in one quarter of OMT
+    static constexpr int l_mult = SEEX;
+    // Cache large enough to store 4 values per OMT for whole overmap for 48 thrity minute intervals assuming 10% of map tiles contain items
+    static constexpr int weather_cache_limit = OMAPX * OMAPY * 4 * 48 / 10;
 
     static const w_point weather_cache_value_not_found = { 0.0, 0.0, 0.0, 0.0, "NFIC", 0, false };
+
+    // Lowers granularity of provided parameters to reduce cache size and increase cache hit rate
```

Comment:
Putting the rounding here will apply to all weather.  That may be what we want, but just to air an alternative, we could just round for the purpose of item rot, and retain higher precision for other things.